utils/manifest-utils/src/keystore/x509_keystore.rs

```diff
@@ -420,21 +420,24 @@ impl X509Keystore {
 
     fn verify_cert<S: AsRef<str>>(&self, cert: S) -> anyhow::Result<PKey<Public>> {
         let cert_chain = Self::decode_cert_chain(cert)?;
+        let cert = match cert_chain.last().map(Clone::clone) {
+            Some(cert) => cert,
+            None => bail!("Unable to verify X509 certificate. No X509 certificate in payload."),
+        };
         let store = self
             .store
             .read()
             .map_err(|err| anyhow::anyhow!("Err: {}", err.to_string()))?;
-        let cert = match cert_chain.last().map(Clone::clone) {
-            Some(cert) => cert,
-            None => bail!("Unable to verify certificate. No certificate."),
-        };
+        if store.store.objects().is_empty() {
+            bail!("Unable to verify X509 certificate. No X509 certificates in keystore.")
+        }
         let mut cert_stack = openssl::stack::Stack::new()?;
         for cert in cert_chain {
             cert_stack.push(cert).unwrap();
         }
         let mut ctx = X509StoreContext::new()?;
         if !(ctx.init(&store.store, &cert, &cert_stack, |ctx| ctx.verify_cert())?) {
-            bail!("Invalid certificate");
+            bail!("Unable to verify X509 certificate.");
```

Comment:
Instead of returning "invalid certificate" I changed it to "Unable to verify X509 certificate", because:
1. I wanted to be clear it is a part of X509 verification logic.
2. Inability to verify certificate happens when Provider has no issuer in keystore. Message "Invalid certificate" was misleading in this context (it could be interpreted as malformed cert content).